libavformat/udp.c

```diff
@@ -1144,6 +1144,10 @@ static int udp_write(URLContext *h, const uint8_t *buf, int size)
     }
 
     if (!s->is_connected) {
+        if (!s->dest_addr_len && !s->dest_addr.ss_family) {
+            ff_udp_get_last_recv_addr(h, &s->dest_addr);
+            s->dest_addr_len = sizeof(s->dest_addr);
+        }
```

Comment:
I think this patch should be sent to the ML, since I'm very much not sure if this is a sensible thing to do.
It changes the behaviour of the udp URLContext in a potentially unexpected way.

Also, I added the addr length in my latest update of the patchset.